junit-platform-commons/src/main/java/org/junit/platform/commons/util/CollectionUtils.java

```diff
@@ -139,7 +143,11 @@ public static boolean isConvertibleToStream(Class<?> type) {
 				|| Iterable.class.isAssignableFrom(type)//
 				|| Iterator.class.isAssignableFrom(type)//
 				|| Object[].class.isAssignableFrom(type)//
-				|| (type.isArray() && type.getComponentType().isPrimitive()));
+				|| (type.isArray() && type.getComponentType().isPrimitive())//
+				|| Arrays.stream(type.getMethods())//
+						.filter(m -> m.getName().equals("iterator"))//
+						.map(Method::getReturnType)//
+						.anyMatch(returnType -> returnType == Iterator.class));
```

Comment:
:question: How does this work? Isn't the return type `kotlin.collections.Iterator`, not `java.util.Iterator`?

Reply:
This works because when Kotlin is used under the Jvm, some types are mapped to Java types. The kotlin.collections-Iterator is one of them and is mapped to java.util.Iterator. See also: https://kotlinlang.org/docs/java-interop.html#mapped-types.

Reply:
No worries. I can imagine you are busy. I'm glad you think it is a good improvement.